Makefile

```
@@ -139,13 +144,113 @@ help-install:
 #
 ###############################################################
 
+### Deployment ################################################
+#
+# General release steps:
+# --------------------------------------------------------------
+#  (1) commit all files and remove any untracked files
+#      `git status` should show nothing.
+#  (2) `make release-changelog` to print a suggested
+#      update to CHANGELOG.md. Replace "XXX" with your
+#      intended tag.
+#  (3) create an annotated tag (with -a or -s)
+#  (4) `make release-prep` which:
+#     - checks the tag
+#     - creates a virtualenv
+#     - builds and installs the sdist
+#  (5) `make release-verify` which:
+#     - runs tests
+#     - builds docker
+#  (6a) if everything succeeds: run `make release-upload`
+#       and execute the commands in order.
+#  (6b) if anything goes wrong, rollback the various steps:
+#     - delete on docker hub
+#     - delete local docker image
+#     - delete tag locally
+#     - make clean
+#
+release-check:
+	@if test -z "$(VERSION)"; then                    \
+		echo VERSION is not set.;                 \
+		echo Please create a git tag;             \
+		exit 100;                                 \
+	elif [[ "$(VERSION)" == *"dirty"* ]] ; then       \
+		echo VERSION is dirty.;                   \
```

```suggestion
		echo Please create a git tag at the current version;             \
```

I did have a tag, but it was not at the current version.

---

Yeah, I added the changelog logic after the fact, so the docs need to be updated.

---

whoops, I meant this for "VERSION is not set."